backend/capellacollab/core/authentication/database/__init__.py

```diff
@@ -19,135 +20,140 @@
 from capellacollab.users.models import Role
 
 
-def verify_admin(token=Depends(JWTBearer()), db=Depends(get_db)):
-    if not is_admin(token, db):
-        raise HTTPException(
-            status_code=403,
-            detail={
-                "reason": "The role administrator is required for this transaction.",
-            },
+class RoleVerification:
+    def __init__(self, required_role: Role, verify: bool = True):
+        self.required_role = required_role
+        self.verify = verify
+
+    def __call__(self, token=Depends(JWTBearer()), db=Depends(get_db)) -> bool:
```

Comment:
The return type is intentional. It allows to use the object to execute checks in the code, e.g. you'd like to find out if a user has specific roles, but you don't want to throw a `HTTPException`. 
Replacing the `verify` flag with exception handling in the calling context is not a good idea imho (because we would need to catch the `HTTPException`).